pkg/util/error.go

```diff
@@ -22,8 +27,35 @@ const (
 
 // WriteError write a go error with the correct status code.
 func WriteError(err error, w http.ResponseWriter) {
+	var connectErr *connect.Error
+	if ok := errors.As(err, &connectErr); ok {
+		writeErr := errorWriter.Write(w, &http.Request{
+			Header: http.Header{"Content-Type": []string{"application/json"}},
+		}, err)
+		if writeErr != nil {
+			http.Error(w, writeErr.Error(), http.StatusInternalServerError)
+		}
+		return
+	}
 	status, cerr := ClientHTTPStatusAndError(err)
-	http.Error(w, cerr.Error(), status)
+	WriteErrorWithStatus(cerr, w, status)
+}
+
+func WriteErrorWithStatus(err error, w http.ResponseWriter, status int) {
```

Comment:
nit: a more natural signature here might be

```suggestion
func WriteErrorWithStatus(w http.ResponseWriter, status int, err error) {
```

since it's much more common in Go to have the writer first, but keeping this signature consistent with `WriteError(err error, w http.ResponseWriter)` makes sense.

Reply:
Yeah I also wonder if we should also move it to `util/http` and do `Errorr..(` for those function names.

so `httputil.Error(w,error)`

Reply:
did the whole refactoring now so we don't have it more painful in the future